### ISSUE TITLE
Log ongoing image pull progress

### ISSUE THREAD

**Issue by simon3z** [openshift/image-inspector#13]
When an image is pulled:

```
2016/05/16 08:34:45 Pulling image docker.io/fedora:latest
...

```

there's no progress report.

We should log progress at constant intervals.

**Comment by simon3z** [comment]
cc @enoodle @pweil-

**Comment by enoodle** [comment]
seems like we will need to use https://github.com/docker/docker/blob/77237be0e23dfaba4d4d97a525c3ecc7fb819042/utils/jsonmessage.go
to get the json we will need to change https://github.com/openshift/image-inspector/blob/master/pkg/inspector/image-inspector.go#L74 to have an output stream and rawJSON.

**Comment by simon3z** [comment]
OK, about how it should look like: every minute we log a line about the progress. (Not a real live progress bar which may break the terminal and the logging applications).

**Comment by enoodle** [comment]
I thought we wanted something like:
```
Trying to pull repository docker.io/efreiber/image-inspector ... 2.0.t: Pulling from efreiber/image-inspector
d36ed9a14740: Pull complete 
e0e3e6aba515: Downloading [===============>                                   ] 21.61 MB/68.7 MB
404f91b55704: Download complete 
44e49beea66e: Download complete 
a75ef532b166: Download complete 
7f9da241a756: Download complete 
e60ef65e2bc9: Download complete 
3aa451972f8e: Download complete 
525776ca32ce: Download complete 
460abc733bed: Download complete 
47ac0dada72b: Download complete 
```
That updated in place in the terminal, similar to what you see when you do ```docker pull ...```. 
AFAIK The feedback that we get from docker is in this form so it will be difficult to know the exact percentage of the image that we already downloaded / extracted, but we can log something like "Pulling..." every minute, is that good enough ?

**Comment by simon3z** [comment]
@enoodle and how that would be coping with kubernetes pods output logging?

**Comment by enoodle** [comment]
@simon3z The first option not so much. The second one ( ```Pulling...``` ) will just show this multiple times in the log.

**Comment by simon3z** [comment]
> @simon3z The first option not so much. The second one ( Pulling... ) will just show this multiple times in the log.

@enoodle that's why I suggested the second one to begin with. I think that's the way to go. I think logging every minute is a good interval (cc @pweil- )

**Comment by pweil-** [comment]
how about we just give the `OutputStream` in the `ImagePullOptions` and point to `os.StdOut`?  It's a simple change and looks like:

```go
imagePullOption := docker.PullImageOptions{
			Repository: i.opts.Image,
			OutputStream: os.Stdout,
		}
```

https://gist.github.com/pweil-/9a6c4f3359d39837c657c2da40513d66

**Comment by enoodle** [comment]
@pweil- Yes, This is what I ment in the first comment but I thought it distorts the terminal/log too much so I thought of adding this parser that docker uses. I would suggest going with just "Pulling..." with no progress bar over that.

**Comment by simon3z** [comment]
> how about we just give the OutputStream in the ImagePullOptions and point to os.StdOut? It's a simple change and looks like:

@pweil- @enoodle if there's an option to customize the output so that it's docker/kubernetes/openshift log-friendly, then it's ok with me.

The information I am interested in is what's the progress of the pull but I don't want a fancy output that is hard to understand over the logs and the console.

We can decide to log by time (e.g. every minute) or by progress (e.g. every 10%?).

**Comment by simon3z** [comment]
Fixed by #16
